### PR TITLE
xontrib-cookiecutter v2

### PR DESCRIPTION
**Pull request by anki-code**
Hi! Thank you for this template! I've made many improvements.

### CODE REVIEW

**Comment by anki-code** [comment]
@scopatz or @AstraLuma please merge.

**Comment by scopatz** [comment]
CC @xonsh/xore - I don't really know cookiecutter well enough to evaluate

**Comment by anki-code** [comment]
@scopatz I've tested it 5 times to write a readme. Merge without worries :)

**Comment by scopatz** [comment]
Sorry, I think a real review is needed.  We try not to merge blindly :wink:

**Comment by anki-code** [comment]
Okay, 5 months it was broken, couple of weeks not a problem. 

But you can just 5 minute test if you don't trust me:
```
pip install cookiecutter
cookiecutter gh:anki-code/xontrib-cookiecutter
# enter snail to all lines
pip install -U xontrib-snail
xontrib load snail
# This is snail!
```

**Comment by scopatz** [comment]
Yeah, I really appreciate all of your effort here!

**Comment by anki-code** [comment]
@AstraLuma Jamie, please take a look. 
I'm going to add the info about this template to all of my xontribs that have 8+ stars.

File: README.md [hunk 1]
```diff
@@ -6,6 +6,18 @@ A <a href="https://github.com/audreyr/cookiecutter">cookiecutter</a> template fo
...
 </p>
 
+## Why use this template?
+
+This template includes good pack of prebuilt files: 
+
+* xontrib promotion instructions in the README 
+* `setup.cfg` file to make and install PyPi package easily 
+* `.gitattributes` file to enables Github syntax highlighting for `*.xsh` files
+* `.gitignore` file with standart list of directories to ignore
```
Comment (AstraLuma) [inline README.md]:
```suggestion
* `.gitignore` file with standard list of directories to ignore
```

**Review by AstraLuma — CHANGES_REQUESTED** [review]

**Comment by anki-code** [comment]
@AstraLuma I fixed your proposal almost immediately. May be the notification missing. Please review.

**Comment by gforsyth** [comment]
@anki-code -- a lot of us have 100s, if not 1000s of github notifications, so we tend to turn off commit-level notifications.  Always good to ping us on an issue after pushing up a commit.

File: xontrib-{{cookiecutter.project_slug}}/setup.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+setuptools.setup(
+    name='xontrib-{{cookiecutter.project_slug}}',
+    version='{{cookiecutter.version}}',
+    license='MIT',
+    author='{{cookiecutter.full_name}}',
+    author_email='{{cookiecutter.email}}',
+    description="{{cookiecutter.project_short_description}}",
+    long_description=long_description,
+    long_description_content_type='text/markdown',
+    python_requires='>=3.5',
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/setup.py:19]:
```suggestion
    python_requires='>=3.6',
```

File: xontrib-{{cookiecutter.project_slug}}/setup.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    platforms='any',
+    url='https://github.com/{{cookiecutter.github_username}}/xontrib-{{cookiecutter.project_slug}}',
+    project_urls={
+        "Documentation": "https://github.com/{{cookiecutter.github_username}}/xontrib-{{cookiecutter.project_slug}}/blob/master/README.md",
+        "Code": "https://github.com/{{cookiecutter.github_username}}/xontrib-{{cookiecutter.project_slug}}",
+        "Issue tracker": "https://github.com/{{cookiecutter.github_username}}/xontrib-{{cookiecutter.project_slug}}/issues",
+    },
+    classifiers=[
+        "Programming Language :: Python :: 3",
+        "Programming Language :: Python :: 3.5",
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/setup.py:33]:
```suggestion
```

File: xontrib-{{cookiecutter.project_slug}}/xontrib/{{cookiecutter.project_slug}}.xsh [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import xonsh
+
+
+__all__ = ()
+
+print('This is {{cookiecutter.project_slug}}!')
+
+# Good strart! Get more documentation -> https://xon.sh/contents.html#guides
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/xontrib/{{cookiecutter.project_slug}}.xsh:8]:
```suggestion
# Good start! Get more documentation -> https://xon.sh/contents.html#guides
```

File: xontrib-{{cookiecutter.project_slug}}/README.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+--------------------
+
+## xontrib promotion (READ and REMOVE THIS SECTION)
+
+After you create the xontrib repository you can do some helpful tasks to spread the words about xontrib.
+
+**Repository name**. It's a good practice to add `xontrib-` prefix before the name of your repository. It allows use the Github search to find it.
+
+**Add topics to the repository**. To show the xontrib repository in Github Topics please add topics `xonsh` and `xontrib` to the repository "About" setting. Also add thematic topics for example `ssh` if your xontrib improves of working with it.
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/README.md:47]:
```suggestion
**Add topics to the repository**. To show the xontrib repository in Github Topics please add topics `xonsh` and `xontrib` to the repository "About" setting. Also add thematic topics, for example,  `ssh` if your xontrib helps work with `ssh`.
```

File: xontrib-{{cookiecutter.project_slug}}/README.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+This package was created with [xontrib cookiecutter template](https://github.com/xonsh/xontrib-cookiecutter).
+
+
+--------------------
+
+## xontrib promotion (READ and REMOVE THIS SECTION)
+
+After you create the xontrib repository you can do some helpful tasks to spread the words about xontrib.
+
+**Repository name**. It's a good practice to add `xontrib-` prefix before the name of your repository. It allows use the Github search to find it.
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/README.md:45]:
```suggestion
**Repository name**. It's a good practice to add `xontrib-` prefix before the name of your repository. It helps Github search find it.
```

File: xontrib-{{cookiecutter.project_slug}}/README.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+## Credits
+
+This package was created with [xontrib cookiecutter template](https://github.com/xonsh/xontrib-cookiecutter).
+
+
+--------------------
+
+## xontrib promotion (READ and REMOVE THIS SECTION)
+
+After you create the xontrib repository you can do some helpful tasks to spread the words about xontrib.
```
Comment (gforsyth) [inline xontrib-{{cookiecutter.project_slug}}/README.md:43]:
```suggestion
After you create the xontrib repository you can do some helpful tasks to spread the word about your xontrib.
```

File: README.md [hunk 8]
```diff
@@ -1,28 +1,56 @@
...
-1. Register to https://pypi.org/ and [create API token](https://pypi.org/help/#apitoken).
-2. Go to repository "Settings" - "Secrets" and add keys `PYPI_USERNAME` and `PYPI_PASSWORD`.
-3. Click "Actions" link on your Github repository.
-4. Click "Set up this workflow" on "Publish Python Action".
-5. Commit the config without any changes.
-6. Now when you create new Release the Github Actions will publish the xontrib to PyPi automatically. Release status will be in Actions sction.
+## Create new xontrib
 
-**Add preview image**. Add the image to repository "Settings" - "Options" - "Social preview". It allows to show preview image in Github Topics and social networks.
+To create xontrib from the template just run:
```
Comment (gforsyth) [inline README.md:23]:
```suggestion
To create a `xontrib` from this template just run:
```

**Review by gforsyth — CHANGES_REQUESTED** [review]
Thanks for putting this together @anki-code!  I think we can remove Python 3.5, and I've tweaked some of the language a little bit.

**Comment by anki-code** [comment]
@gforsyth many thank! All changes are merged.

> Always good to ping us on an issue after pushing up a commit.

Got it! Ready to merge.

**Review by gforsyth — APPROVED** [review]
Ok, this looks good to me.  Thanks @anki-code !

**Pull request merged**